lib/db2.js

```diff
@@ -45,6 +45,9 @@ function DB2(settings) {
   var dsn = settings.dsn;
   if (dsn) {
     this.connStr = dsn;
+    if (!('CurrentSchema' in parseDSN(dsn))) {
```

Comment:
I'm not sure I understand this change yet.  If CurentSchema is not part of DSN you grab UID as the CurrentSchema and add it to the connection string.  You don't however set this.schema = parseDSN(dsn).UID.  Also, there should be an else block where you set this.schema = parseDSN(dsn).CurrentSchema in the event that it is included in the DSN.  Otherwise we will exit this if block and never have this.schema set.